### PR TITLE
Droth 355 fetching complementary geometry from vvh

### PR DESCRIPTION
**Pull request by marquesfm**

### CODE REVIEW

**Comment by tvenhola** [comment]
Looks good to me. Merging master for conflict resolution.

**Comment by marquesfm** [comment]
Good to me.

File: digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/asset/asset.scala [hunk 1]
```diff
@@ -4,6 +4,23 @@ import fi.liikennevirasto.digiroad2.{Point, Vector3d}
...
+sealed trait LinkGeomSource{
+  def value: Int
+}
+
+object LinkGeomSource{
+  val values = Set(NormalLinkInterface, ComplimentaryLinkInterface , SuravageLinkInterface, FrozenLinkInterface, HistoryLinkInterface)
+
+  def apply(intValue: Int): LinkGeomSource = values.find(_.value == intValue).getOrElse(Unknown)
+
+  case object NormalLinkInterface extends LinkGeomSource {def value = 1;}
```
Comment (halied) [inline digiroad2-geo/src/main/scala/fi/liikennevirasto/digiroad2/asset/asset.scala]: Remove semicolons?

**Comment by halied** [comment]
Ok by me

**Comment by oappicgi** [comment]
Direct link to complementary link gives error to browser console. Would be nice if it didn't (example:
http://localhost:9001/index.html#linkProperty/500023737

otherwise ok

**Comment by tvenhola** [comment]
In Viite this ^ has been solved. We could steal it.

**Pull request merged**